sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/generators.scala

```diff
@@ -146,7 +146,8 @@ case class Stack(children: Seq[Expression]) extends Generator {
     } else {
       for (i <- 1 until children.length) {
         val j = (i - 1) % numFields
-        if (children(i).dataType != elementSchema.fields(j).dataType) {
+        if (children(i).dataType != NullType &&
```

Comment:
can you verify the behavior of hive? Does hive accept int and long type in the same column?